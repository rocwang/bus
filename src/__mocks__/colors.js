export default {
  oceanBlue: "#001120",
  green: "#8fc240",
  blue: "#00a3da",
  yellow: "#ffdb1b",
  red: "#da2f2d",
  orange: "#ea8c2e",
  black: "#000",
  grayA: "#333",
  grayB: "#666",
  grayC: "#999",
  grayD: "#ccc",
  white: "#fff",
};
