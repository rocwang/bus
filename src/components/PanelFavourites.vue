<template>
  <Panel :isCollapsed="isCollapsed">
    <template v-slot:icon>
      <RoundIconStarFull />
    </template>
    <template v-slot:title>
      Favourites
    </template>
    <template v-slot:buttons>
      <Buttonizer modifier="icon">
        <button aria-label="edit">
          <IconEdit />
        </button>
        <button @click="handleArrowClick" aria-label="toggle the panel">
          <IconArrow :direction="isCollapsed ? 'up' : 'down'" />
        </button>
      </Buttonizer>
    </template>

    <template v-slot:body>
      <ul>
        <li
          v-for="item in items"
          :key="`${item.stopCode}-${item.routeShortName}`"
        >
          {{ item.name }}
        </li>
      </ul>
    </template>
  </Panel>
</template>

<script>
import Panel from "../components/Panel";
import RoundIconStarFull from "../components/RoundIconStarFull";
import IconArrow from "../components/IconArrow";
import IconEdit from "../components/IconEdit";
import Buttonizer from "../components/Buttonizer";
import { list } from "../favouritesStore";

export default {
  name: "PanelFavourites",
  components: { Panel, RoundIconStarFull, IconArrow, IconEdit, Buttonizer },
  props: {
    isCollapsedInitially: {
      type: Boolean,
      default: true
    }
  },
  data() {
    return {
      isCollapsed: this.isCollapsedInitially,
      items: []
    };
  },
  async created() {
    this.items = await list();
  },
  methods: {
    handleArrowClick() {
      this.isCollapsed = !this.isCollapsed;
    }
  }
};
</script>
