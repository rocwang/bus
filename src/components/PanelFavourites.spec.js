import { mount } from "@vue/test-utils";
import PanelFavourites from "./PanelFavourites";
import colors from "./__fixtures__/colors";

describe("PanelFavourites.vue", () => {
  it("matches the snapshot", () => {
    const wrapper = mount(PanelFavourites, {
      stubs: ["router-link"],
      provide: {
        colors
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("can be expanded initially", () => {
    const wrapper = mount(PanelFavourites, {
      stubs: ["router-link"],
      provide: {
        colors
      },
      propsData: {
        isCollapsedInitially: false
      }
    });
    expect(wrapper.vm.isCollapsed).toBe(false);
  });

  it("renders favourites", () => {
    const wrapper = mount(PanelFavourites, {
      stubs: ["router-link"],
      provide: {
        colors
      },
      propsData: {
        favouritesWithTrips: [
          { name: "test1", stopCode: "stopCode1", routeShortName: "NX1" },
          { name: "test2", stopCode: "stopCode2", routeShortName: "NX2" }
        ]
      }
    });

    expect(wrapper.findAll("li").length).toBe(2);
  });

  it("can be toggled", () => {
    const wrapper = mount(PanelFavourites, {
      stubs: ["router-link"],
      provide: {
        colors
      }
    });
    const toggleButon = wrapper.find('button[aria-label="toggle the panel"]');

    toggleButon.trigger("click");
    expect(wrapper.vm.isCollapsed).toBe(false);
    toggleButon.trigger("click");
    expect(wrapper.vm.isCollapsed).toBe(true);
  });
});
