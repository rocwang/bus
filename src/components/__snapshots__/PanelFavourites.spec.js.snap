// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelFavourites.vue matches the snapshot 1`] = `
<div
  class="root collapsed"
>
  <div
    class="head"
  >
    <div
      class="icon"
    >
      <svg
        viewBox="0 0 50 50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25 0a25 25 0 1 0 25 24.984C49.991 11.184 38.801 0 25 0"
          fill="#ffdb1b"
        />
         
        <path
          d="M23.39 11.044L19.73 18.8l-8.193 1.248c-1.469.223-2.057 2.115-.992 3.2l5.927 6.034-1.402 8.524c-.252 1.54 1.301 2.695 2.602 1.974L25 35.755l7.329 4.025c1.3.715 2.854-.433 2.602-1.974l-1.402-8.524 5.927-6.035c1.065-1.084.477-2.976-.992-3.199L30.27 18.8l-3.662-7.756a1.763 1.763 0 0 0-3.218 0z"
          fill="#001120"
        />
      </svg>
    </div>
     
    <h1
      class="title"
    >
      <small
        class="prefix"
      />
       
    Favourites
  
    </h1>
     
    <div
      class="buttons"
    >
      <button
        aria-label="edit"
        class=""
      >
        <svg
          viewBox="0 0 30 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.969 4.875l4.698 5.285a.625.625 0 0 1 0 .809L14.292 23.766l-4.834.603c-.646.082-1.192-.533-1.12-1.26l.537-5.437L20.25 4.875a.47.47 0 0 1 .719 0zm8.437-1.342L26.865.673c-.792-.89-2.079-.89-2.875 0l-1.844 2.075a.625.625 0 0 0 0 .809l4.698 5.285a.47.47 0 0 0 .718 0l1.844-2.074c.792-.897.792-2.344 0-3.235zM20 20.285v5.965H3.333V7.5h11.97a.606.606 0 0 0 .442-.205l2.083-2.344c.396-.445.115-1.201-.443-1.201H2.5C1.12 3.75 0 5.01 0 6.563v20.625C0 28.74 1.12 30 2.5 30h18.333c1.38 0 2.5-1.26 2.5-2.813v-9.246c0-.627-.672-.937-1.067-.498l-2.084 2.344a.77.77 0 0 0-.182.498z"
            fill="#ffdb1b"
          />
        </svg>
      </button>
       
      <button
        aria-label="toggle the panel"
        class=""
      >
        <svg
          viewBox="0 0 30 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.01 17.245L.484 15.758a1.572 1.572 0 0 1 0-2.27L13.835.472c.646-.63 1.69-.63 2.33 0l13.35 13.01c.646.63.646 1.647 0 2.27l-1.525 1.486a1.681 1.681 0 0 1-2.357-.027L17.752 9.15v19.243c0 .89-.735 1.607-1.65 1.607h-2.198c-.914 0-1.65-.716-1.65-1.607V9.15l-7.887 8.068a1.67 1.67 0 0 1-2.357.027z"
            fill="#ffdb1b"
          />
        </svg>
      </button>
    </div>
     
    <p
      class="subtitle"
    />
  </div>
   
  <div
    class="body"
  >
    <ul />
  </div>
</div>
`;
